### PR TITLE
feat: support `localePattern` option for locale structured with directory

### PR DESCRIPTION
**Pull request by kazupon**

### CODE REVIEW

**Review by ota-meshi — APPROVED** [review]

**Review by ota-meshi — CHANGES_REQUESTED** [review]
I made a mistake the status.

File: lib/types/settings.ts [hunk 1]
```diff
@@ -29,4 +31,12 @@ export interface SettingsVueI18nLocaleDirObject {
...
    */
   localeKey: LocaleKeyType
+  /**
+   * Specifies how to determine pattern the locale for localization messages.
+   *
+   * This option means, when `localeKey` is `'path'`, you will need to capture the locale using a regular expression.
+   * You need to use the locale capture as a named capture `?<locale>`, so it’s be able to capture from the path of the locale resources.
+   * If you omit it, it will be captured from the resource path with the same regular expression pattern as `vue-cli-plugin-i18n`.
+   */
+  localePattern?: RegExp
```
[the remark this answers is not in the document]
Comment (kazupon) [reply lib/types/settings.ts:41]: Oh, Indeed 😅 
we have to support cases in other formats but  `.eslintrc.js`, such as `.eslintrc.yaml`.

**Comment by kazupon** [comment]
I've just fixed it!

**Review by ota-meshi — APPROVED** [review]
LGTM!

**Pull request merged**